src/besselj.jl

```diff
@@ -31,6 +51,19 @@ function besselj0(x::Float64)
         xn = x - PIO4(T)
         p = p * cos(xn) - w * q * sin(xn)
         return p * SQ2OPI(T) / sqrt(x)
+    else
+        xinv = inv(x)
+        x2 = xinv*xinv
+
+        p = (one(T), -1/16, 53/512, -4447/8192, 5066403/524288)
+        p = evalpoly(x2, p)
+        a = SQ2OPI(T) * sqrt(xinv) * p
+
+        q = (-1/8, 25/384, -1073/5120, 375733/229376, -55384775/2359296)
+        xn = fma(xinv, evalpoly(x2, q), - PIO4(T))
+        b = cos(x)*cos(xn) - sin(x)*sin(xn)#cos(x + xn)
+        #b = cos(x + xn)
```

Comment:
These are the two lines that need some care. `b = cos(x)*cos(xn) - sin(x)*sin(xn)` is very accurate but slow.... 
`b = cos(x + xn)` is fast but less accurate

Reply:
Can you rephrase using `cospi`/`sinpi`? Those work better for big inputs.

Reply:
Hmmm ya I think this needs some type of reduction. The issue is that `x` is really large and `xn` is small and negative so we lose precision making that addition before the `cosine`. Whereas `cos(x)*cos(xn)` makes that reduction for us.

Do you mean like `b = cospi(x/pi + xn/pi)`? Or something different

Reply:
```julia
c1 = sincos(x)
c2 = sincos(xn)
b = c1[2]*c2[2] - c1[1]*c2[1]
```
This is just as fast as the old versions and more accurate.... but still 80% slower than `b = cos(x + xn)`

Reply:
There's a much better way to do this. Use
```
mycos(x, xn)
    n, r = Base.Math.rem_pio2_kernel(x)
    hi, lo = r.hi, r.lo
    n2, r = Base.Math.rem_pio2_kernel(hi + (r + lo))
    n = (n+n2)&3
    if n == 0
        return cos_kernel(y)
    elseif n == 1
        return -sin_kernel(y)
    elseif n == 2
        return -cos_kernel(y)
    else
        return sin_kernel(y)
    end
end
```

Reply:
I'm getting several errors running this. Is xn supposed to be y?
Also    ` n2, r = Base.Math.rem_pio2_kernel(hi + (r + lo))` return a method error for me (r is some `Base.Math.DoubleFloat64`)

Reply:
oops, fixed.

Reply:
Ok works now but gives worse errors...
<img width="641" alt="Screen Shot 2022-02-16 at 4 00 13 PM" src="https://user-images.githubusercontent.com/57471570/154364569-fa80077b-4371-4577-afcc-1f7a6b50f8a5.png">

Reply:
```julia
julia> cos(big"100.0" + big"-0.10")
0.8074586576995466543840048046567971916342575898589744635569386441759010692980412

julia> mycos(100.0, -0.10)
0.8074586576995554

julia> cos(100.0 + (-0.10))
0.80745865769955
```

Reply:
Here's a version that gets within about 1 ulp (where tested, there's probably a case where it does slightly worse).
```
function mycos(x, xn)
    s = x + xn
    n, r = Base.Math.rem_pio2_kernel(s)
    lo = r.lo - ((s - x) - xn)
    hi = r.hi + lo
    y = Base.Math.DoubleFloat64(hi, r.hi-hi+lo)
    n = n&3
    if n == 0
        return Base.Math.cos_kernel(y)
    elseif n == 1
        return -Base.Math.sin_kernel(y)
    elseif n == 2
        return -Base.Math.cos_kernel(y)
    else
        return Base.Math.sin_kernel(y)
    end
end
```

Reply:
Some analysis suggests that this should be extremely accurate as long as `abs(x+xn) < 10^14`

Reply:
Yes!! This looks great. Awesome stuff and is really fast.. This is now 2.8x faster than SpecialFunctions.jl and really accurate!
<img width="636" alt="Screen Shot 2022-02-17 at 12 14 41 AM" src="https://user-images.githubusercontent.com/57471570/154416530-a5c7b69c-f23a-49d7-bbee-bc629c9d248f.png">